servicetalk-http-netty/src/test/java/io/servicetalk/http/netty/StreamObserverTest.java

```diff
@@ -162,15 +162,16 @@ public void maxActiveStreamsViolationError() throws Exception {
 
             ExecutionException e = assertThrows(ExecutionException.class,
                     () -> connection.request(connection.get("/second")).toFuture().get());
-            assertThat(e.getCause(), instanceOf(StreamException.class));
+            assertThat(e.getCause(), instanceOf(Http2Exception.class));
+            assertThat(e.getCause().getCause(), instanceOf(StreamException.class));
 
             verify(clientMultiplexedObserver, times(2)).onNewStream();
             verify(clientStreamObserver, times(2)).streamEstablished();
             verify(clientDataObserver, times(2)).onNewRead();
             verify(clientDataObserver, times(2)).onNewWrite();
             verify(clientReadObserver).readCancelled();
-            verify(clientWriteObserver).writeFailed(e.getCause());
-            verify(clientStreamObserver, await()).streamClosed(e.getCause());
+            verify(clientWriteObserver).writeFailed(e.getCause().getCause());
+            verify(clientStreamObserver, await()).streamClosed(e.getCause().getCause());
```

Comment:
`WriteObserver` and `StreamObserver` report the unwrapped netty's exception because the reporting happens in `WriteStreamSubscriber`:
https://github.com/apple/servicetalk/blob/c3ff7bf27f461d6a138f48c784e3ceab8678bccb/servicetalk-transport-netty-internal/src/main/java/io/servicetalk/transport/netty/internal/WriteStreamSubscriber.java#L478-L480

The `Single<StreamingHttpResponse>` sees wrapped `IOException`.
@Scottmitch wdyt, should we enrich the error in `WriteStreamSubscriber`?

Reply:
if we are going to wrap the exception seems like we should wrap it everywhere it is exposed.